### PR TITLE
Move the array of plugin names to inside the last if block in VolumePluginMgr#FindPluginBySpec

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In VolumePluginMgr#FindPluginBySpec, we use two arrays, one for VolumePlugin and one for plugin names.

This PR moves the array of plugin names to inside the last if block.

I also release the lock early - before the two if blocks on matches check.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
The pull-kubernetes-dependencies failure was not related to my PR.

**Comment by tedyu** [comment]
ping @msau42 for review

File: pkg/volume/plugins.go [hunk 1]
```diff
@@ -641,33 +641,35 @@ func (pm *VolumePluginMgr) initProbedPlugin(probedPlugin VolumePlugin) error {
...
 	}
 
 	pm.refreshProbedPlugins()
 	for _, plugin := range pm.probedPlugins {
 		if plugin.CanSupport(spec) {
-			matchedPluginNames = append(matchedPluginNames, plugin.GetPluginName())
 			matches = append(matches, plugin)
 		}
 	}
+	pm.mutex.Unlock()
```
Comment (smarterclayton) [inline pkg/volume/plugins.go]: Generally this is considered unacceptable in our style.  A panic in CanSupport would leave the plugin wedged.

If you want to do this, reorganize into two sets of code and continue to use defer

/hold

Comment (tedyu) [reply pkg/volume/plugins.go]: @smarterclayton 
I have restored the lock deferring.

**Comment by tedyu** [comment]
@smarterclayton 
Can you take another look ?

Thanks

**Comment by tedyu** [comment]
gentle ping @msau42 @smarterclayton for review

**Comment by tedyu** [comment]
ping @smarterclayton for review

**Comment by tedyu** [comment]
gentle ping @msau42 @smarterclayton for review

**Comment by tedyu** [comment]
ping @msau42 for review

**Comment by msau42** [comment]
/lgtm
/approve

Please squash your commits, thanks!

**Comment by tedyu** [comment]
There was more recent change to pkg/volume/plugins.go since the creation of the underlying branch.

close in favor of #77838

**Pull request closed without merging**